ember-server/src/test/scala/org/http4s/ember/server/EmberServerSuite.scala

```diff
@@ -92,4 +92,30 @@ class EmberServerSuite extends Http4sSuite {
         r2 <- client.fetchAs[String](request)
       } yield assertEquals(expected, r1) && assertEquals(expected, r2)
   }
+
+  fixture().test("empty stream") {
+    case (server, _) =>
+      import org.http4s.dsl.io._
+      import org.http4s.client.dsl.io._
+      import scala.concurrent.duration._
+
+      val body: Stream[IO, Byte] =
+        Stream.emits(Seq("hello")).through(fs2.text.utf8Encode).covary[IO]
+
+      val uri = Uri
+        .fromString(s"http://${server.address.getHostName}:${server.address.getPort}/echo")
+        .toOption
+        .get
+      val request = GET(body, uri)
+      EmberClientBuilder.default[IO].withIdleConnectionTime(1.seconds).build.use { client =>
```

Comment:
I believe this is the opposite of what we want. we are interesting in seeing what happens when the client timeout is _longer_ than the server timeout i.e. what happens if the server closes their end of the connection and the client try to make another request

Reply:
Also the right client configuration option to use here is `withIdleTimeInPool` -- idle connection time controls how much time the client will wait for a response after a request has been sent before timeout

Reply:
Ugh, actually, I don't think this will test exactly what we want. At worst this will induce a connection reset by peer error which isn't what we want. I think we should just merge in the fix for now and think about how to do the test later, since none of the servers we have available let us emulate that behavior

Reply:
@rossabaker Updated the PR.